src/vector/backends/numpy.py

```diff
@@ -972,6 +981,11 @@ def __array_finalize__(self, obj: typing.Any) -> None:  # type: ignore[override]
         if obj is None:
             return
 
+        if not _is_type_safe(numpy.array(self.tolist())):
+            raise TypeError(
+                "a coordinate must be of the type numpy.integer or numpy.floating"
+            )
```

Comment:
This (and following) is still doing an expensive conversion of a NumPy array to a Python list (and back again) to check for type safety. Anywhere that you have `tolist()` is a red flag.